alma/tests/unit/lib/API/EligibilityHelperTest.php

```diff
@@ -0,0 +1,191 @@
+                        [
+                            'purchase_amount' => 5099,
+                            'installments_count' => 3,
+                            'deferred_days' => 0,
+                            'deferred_months' => 0
+                        ]
+                    ],
+                    'shipping_address' => [
+                        'country' => 'FR'
+                    ],
+                    'billing_address' => [
+                        'country' => 'FR'
+                    ],
+                    'locale' => 'en'
+                ], 
+                //data expected
+                [
+                    $expectedAmountEligibe
+                ]
+            ],
+            'purchase amount not eligible in p3x' => [
+                //data
+                [
+                    'purchase_amount' => 2600,
+                    'queries' => [
+                        [
+                            'purchase_amount' => 2600,
+                            'installments_count' => 3,
+                            'deferred_days' => 0,
+                            'deferred_months' => 0
+                        ]
+                    ],
+                    'shipping_address' => [
+                        'country' => 'FR'
+                    ],
+                    'billing_address' => [
+                        'country' => 'FR'
+                    ],
+                    'locale' => 'en'
+                ], 
+                //data expected
+                [
+                    $expectedAmountNotEligibe
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider getEligibilityData
+     * @return void
+     */
+    public function testEligibilityCheck($data, $expected)
+    {
+        $clientMock = Mockery::mock(ClientHelper::class);
+        $clientMock->shouldReceive('defaultInstance')->andReturn($_ENV['ALMA_API_KEY']);
+        $clientMock->shouldReceive('createInstance');
+
+        $eligibilityHelper = Mockery::mock(EligibilityHelper::class)->shouldAllowMockingProtectedMethods()->makePartial();
+
+        $contextMock = Mockery::mock(Context::class);
+        $contextMock->cart = Mockery::mock(Cart::class);
+        $contextMock->cart->shouldReceive('getOrderTotal')->andReturn(almaPriceFromCents($data['purchase_amount']));
+        $contextMock->cart->id_address_delivery = 5;
+        $contextMock->cart->id_address_invoice = 5;
+        $contextMock->language = Mockery::mock(Language::class);
+        $contextMock->language->iso_code = 'en';
+
+        $paymentDataMock = Mockery::mock(PaymentData::class);
+        $paymentDataMock->shouldReceive('dataFromCart')->andReturn($data);
+
+        $eligibility = $eligibilityHelper->eligibilityCheck($contextMock);
+        $expectedPaymentPlan = $expected[0]->getPaymentPlan();
+        $eligibilityPaymentPlan = $eligibility[0]->getPaymentPlan();
+        if (count($eligibilityPaymentPlan) > 0) {
```

Comment:
```suggestion
        $this->assertEquals(0, count($eligibilityPaymentPlan));
```

Is it really possible to not have paymentPlan ?
Maybe you should test this with phpunit ?

I am not sure about that, it is only a suggestion.

Reply:
@syjust-alma Yes, if the installment is not eligible, there are no paymentPlan.
But I don't understand your suggestion sorry.

Reply:
not testable

Reply:
@syjust-alma we replace unit test in integration test